src/call_reply.c

```diff
@@ -0,0 +1,503 @@
+ *
+ * The returned values are only borrowed and their lifetime is
+ * as long as the given CallReply. In addition there is no need
+ * to manually free the returned CallReplies, it will be freed when
+ * the root CallReplied will be freed.
+ */
+int callReplyGetAttributeElement(CallReply *rep, size_t idx, CallReply **key, CallReply **val) {
+    return callReplyGetMapElementInternal(rep, idx, key, val, REDISMODULE_REPLY_MAP);
+}
+
+/**
+ * Return big number reply value, applicabale for:
+ * * REDISMODULE_REPLY_BIG_NUMBER
+ *
+ * The returned value is only borrowed and its lifetime is
+ * as long as the given CallReply.
+ * The returned value is promised to be a big number as describe
+ * on RESP3 spacifications.
+ * The returned value is not NULL terminated and its mandatory to
+ * give the len argument
+ */
+const char* callReplyGetBigNumber(CallReply *rep, size_t *len) {
+    callReplyParse(rep);
+    if (rep->type != REDISMODULE_REPLY_BIG_NUMBER) return NULL;
+    *len = rep->len;
+    return rep->val.str;
+}
+
+/**
+ * Return verbatim string reply value, applicabale for:
+ * * REDISMODULE_REPLY_VERBATIM_STRING
+ *
+ * An optional output argument can be given to get verbatim reply
+ * format, or NULL if not needed.
+ *
+ * The returned value is only borrowed and its lifetime is
+ * as long as the given CallReply.
+ * The returned value is not NULL terminated and its mandatory to
+ * give the len argument
+ */
+const char* callReplyGetVerbatim(CallReply *rep, size_t *len, const char **format){
+    callReplyParse(rep);
+    if (rep->type != REDISMODULE_REPLY_VERBATIM_STRING) return NULL;
+    *len = rep->len;
+    if (format) *format = rep->val.verbatim_str.format;
+    return rep->val.verbatim_str.str;
+}
+
+/**
+ * Return the current reply blob. The return value is borrowed
+ * and can only be used as long as the CallReply is alive
+ *
+ * The returned value is only borrowed and its lifetime is
+ * as long as the given CallReply.
+ */
+const char* callReplyGetProto(CallReply *rep, size_t *proto_len) {
+    *proto_len = rep->proto_len;
+    return rep->proto;
+}
+
+/**
+ * Return CallReply private data as it was give when the reply was
+ * created using callReplyCreate
+ */
+void* callReplyGetPrivateData(CallReply *rep) {
+    return rep->private_data;
+}
+
+/**
+ * Return true if the reply or one of its children format is resp3
+ */
+int callReplyIsResp3(CallReply *rep) {
+    return rep->flags & REPLY_FLAG_RESP3;
+}
+
+/**
+ * Create a new CallReply struct from the give reply blob.
+ * The function takes ownership on the reply blob which means
+ * that it should not be used after calling this function.
+ * The reply blob will be freed when the returned CallReply
+ * object will be freed using freeCallReply.
+ *
+ * The given private_data can be retriv from the
+ * returned CallReply object or any of its children (in case
+ * of nested reply) using callReplyGetPrivateData,
+ */
```

Comment:
```suggestion
/* Create a new CallReply struct from the give reply blob.
 * The function takes ownership over the reply blob, which means
 * that it should not be used after calling this function.
 * The reply blob will be freed when the returned CallReply
 * object is freed using freeCallReply.
 *
 * The given private_data can be retrieved from the
 * returned CallReply object or any of its children (in case
 * of a nested reply) using callReplyGetPrivateData. */
```